docs/examples/log4net.rst

```diff
@@ -32,24 +32,41 @@ Here's a sample module that configures Autofac to inject ``ILog`` parameters bas
 
       private static void OnComponentPreparing(object sender, PreparingEventArgs e)
       {
-        e.Parameters = e.Parameters.Union(
-          new[]
-          {
-            new ResolvedParameter( 
-                (p, i) => p.ParameterType == typeof(ILog), 
-                (p, i) => LogManager.GetLogger(p.Member.DeclaringType)
-            ),
-          });
+        e.Parameters = e.Parameters.Union(new[]
+        {
+          new ResolvedParameter(
+            (p, i) => p.ParameterType == typeof(ILog),
+            (p, i) => LogManager.GetLogger(p.Member.DeclaringType))
+        });
+      }
+
+      protected override void Load(ContainerBuilder builder)
+      {
+        // Handle registrations in nested lifetime scopes.
+        builder.RegisterBuildCallback(c => c.ChildLifetimeScopeBeginning += OnChildLifetimeScopeBeginning);
```

Comment:
This should register for the container creating child scopes but I don't think it handles those children creating scopes, like...

```csharp
using(var scope1 = container.BeginLifetimeScope())
{
  // scope1 should have things attached
  using(var scope2 = scope1.BeginLifetimeScope())
  {
    // I don't think scope2 will be registered
  }
}
```

One of the reasons the example isn't super robust is... it's an example, and we don't have unit tests or anything to verify that a super rich example like this actually handles everything.

Additional related issues:

- autofac/Autofac#218 - It might be better to figure out how to properly handle nested lifetime scopes than enhance this example.
- autofac/Autofac#912 - If this updated example does work, it may be because of this bug where nested container builders incorrectly share the build callback list.

Reply:
You are right - it wasn't handling children scopes. I've updated PR to handle that.